test/test-helpers.js

```diff
@@ -0,0 +1,31 @@
+import assert from 'assert'
+
+/**
+ * @param {object} args
+ * @param {any} [args.responseBody]
+ * @param {number} [args.status=200]
+ * @param {boolean} [args.throwError=false]
+ * @param {string} [args.errorMessage]
+ * @returns {{requestedUrls: string[], mockFetch: typeof globalThis.fetch}}
+ */
+export function createMockedFetch ({ responseBody, status, throwError, errorMessage }) {
+  /** @type {string[]} */
+  const requestedUrls = []
+
+  /** @type {typeof globalThis.fetch} */
+  const mockFetch = async (url, allOpts) => {
+    assert(typeof url === 'string')
+    requestedUrls.push(url.toString())
+
+    if (throwError) {
+      throw new Error(errorMessage)
+    }
+
+    return new Response(JSON.stringify(responseBody), {
+      status,
```

Comment:
```suggestion
      status ?? 200,
```

So that we can simplify usage:

```js
const { requestedUrls, mockFetch } = createMockedFetch({ responseBody: [{ total: '100', successful: '95' }] })